src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs

```diff
@@ -639,6 +684,594 @@ public static void ReconcileVectorKeyValues(IHostEnvironment env, IDataView[] vi
+            ch.AssertValue(row);
+            ch.Assert(Utils.Size(getters) == schema.ColumnCount);
+            ch.Assert(Utils.Size(vBufferGetters) == schema.ColumnCount);
+
+            // Get the names of the metrics. For R8 valued columns the metric name is the column name. For R8 vector valued columns
+            // the names of the metrics are the column name, followed by the slot name if it exists, or "Label_i" if it doesn't. 
+            VBuffer<DvText> names = default(VBuffer<DvText>);
+            int metricCount = 0;
+            var metricNames = new List<string>();
+            for (int i = 0; i < schema.ColumnCount; i++)
+            {
+                if (schema.IsHidden(i) || ignoreCol(i))
+                    continue;
+
+                var type = schema.GetColumnType(i);
+                var metricName = row.Schema.GetColumnName(i);
+                if (type.IsNumber)
+                {
+                    getters[i] = RowCursorUtils.GetGetterAs<double>(NumberType.R8, row, i);
+                    metricNames.Add(metricName);
+                    metricCount++;
+                }
+                else if (type.IsVector && type.ItemType == NumberType.R8)
+                {
+                    if (type.VectorSize == 0)
+                    {
+                        ch.Warning("Vector metric '{0}' has different lengths in different folds and will not be averaged for overall results.", metricName);
+                        continue;
+                    }
+
+                    vBufferGetters[i] = row.GetGetter<VBuffer<double>>(i);
+                    metricCount += type.VectorSize;
+                    var slotNamesType = schema.GetMetadataTypeOrNull(MetadataUtils.Kinds.SlotNames, i);
+                    if (slotNamesType != null && slotNamesType.VectorSize == type.VectorSize && slotNamesType.ItemType.IsText)
+                        schema.GetMetadata(MetadataUtils.Kinds.SlotNames, i, ref names);
+                    else
+                    {
+                        var namesArray = names.Values;
+                        if (Utils.Size(namesArray) < type.VectorSize)
+                            namesArray = new DvText[type.VectorSize];
+                        for (int j = 0; j < type.VectorSize; j++)
+                            namesArray[j] = new DvText(string.Format("Label_{0}", j));
+                        names = new VBuffer<DvText>(type.VectorSize, namesArray);
+                    }
+                    foreach (var name in names.Items(all: true))
+                        metricNames.Add(string.Format("{0} {1}", metricName, name.Value));
+                }
+            }
+            ch.Assert(metricNames.Count == metricCount);
+            return metricNames;
+        }
+
+        internal static void GetOverallMetricsData(IHostEnvironment env, IDataView data, int numFolds,
+            out double[] sumMetrics, out double[] sumSqMetrics, out double[] sumWeightedMetrics, out double[] sumSqWeightedMetrics,
```

Comment:
>sumMetrics, out double[] sumSqMetrics, out double[] sumWeightedMetrics [](start = 25, length = 70)

Create a struct/class with metric/weighted metric/metric name, and have the out variable be an array of these structs.